src/Grids/zeros_and_ones.jl

```diff
@@ -4,7 +4,7 @@ using Oceananigans.Architectures: CPU, GPU, AbstractArchitecture
 import Base: zeros
 
 zeros(FT, ::CPU, N...) = zeros(FT, N...)
-zeros(FT, ::GPU, N...) = CUDA.zeros(FT, N...)
+zeros(FT, ::CUDAGPU, N...) = CUDA.zeros(FT, N...)
```

Comment:
would be nice to have KernelAbstraction's allocation method here instead of having to put AMD's `zeros` in an extension.

see:
https://github.com/JuliaGPU/KernelAbstractions.jl/blob/a85cf4958958aa29ed47bf20e532c1e040dc0433/examples/matmul.jl#L30

Reply:
I'd need to think a bit on this given that Julia is still fairly new to me - I'm not seeing how we would pull the backend architecture required as input for the `KernelAbstractions.zeros` api in this context.

Reply:
we have a `device` function in `Architectures.jl` that grabs the KA backend so this would probably suffice
```julia
zeros(FT, arch::AbstractArchitecture, N...) = KA.zeros(device(arch), FT, N...)
```

Reply:
Added this change in https://github.com/CliMA/Oceananigans.jl/pull/3475/commits/aa6a3aa776da5e1d693ad5c858ad3d3678b5eee7

Only note here is the following warning thrown during precompilation
```
  1 dependency had output during precompilation:
┌ Oceananigans
│  WARNING: using KernelAbstractions.GPU in module Grids conflicts with an existing identifier.
│  WARNING: using KernelAbstractions.CPU in module Grids conflicts with an existing identifier.
└  
```